src/components/Earn.jsx

```diff
@@ -290,135 +317,145 @@ export default function Earn() {
+
           {!serviceInfo?.coinjoinInProgress && (
-            <rb.Form onSubmit={onSubmit} validated={validated} noValidate>
-              {!serviceInfo?.makerRunning && !isWaitingMakerStart && !isWaitingMakerStop && (
-                <>
-                  {settings.useAdvancedWalletMode && (
-                    <rb.Form.Group className="mb-3" controlId="offertype">
-                      <ToggleSwitch
-                        label={t('earn.toggle_rel_offer')}
-                        initialValue={isRelOffer}
-                        onToggle={(isToggled) => setAndPersistOffertype(isToggled ? OFFERTYPE_REL : OFFERTYPE_ABS)}
-                        disabled={isLoading}
-                      />
-                    </rb.Form.Group>
-                  )}
-                  {isRelOffer ? (
-                    <rb.Form.Group className="mb-3" controlId="feeRel">
-                      <rb.Form.Label className="mb-0">
-                        {t('earn.label_rel_fee', {
-                          fee: feeRel !== '' ? `(${factorToPercentage(feeRel)}%)` : '',
-                        })}
-                      </rb.Form.Label>
-                      <div className="mb-2">
-                        <rb.Form.Text className="text-secondary">{t('earn.description_rel_fee')}</rb.Form.Text>
-                      </div>
-                      {isLoading ? (
-                        <rb.Placeholder as="div" animation="wave">
-                          <rb.Placeholder xs={12} className={styles['input-loader']} />
-                        </rb.Placeholder>
-                      ) : (
-                        <rb.Form.Control
-                          type="number"
-                          name="feeRel"
-                          value={factorToPercentage(feeRel)}
-                          className="slashed-zeroes"
-                          min={feeRelPercentageMin}
-                          max={feeRelPercentageMax}
-                          step={feeRelPercentageStep}
-                          required
-                          onChange={(e) => setAndPersistFeeRel(percentageToFactor(e.target.value))}
-                        />
+            <Formik initialValues={initialValues} validate={validate} onSubmit={onSubmit}>
+              {({ handleSubmit, setFieldValue, handleChange, handleBlur, values, touched, errors, isSubmitting }) => (
+                <rb.Form onSubmit={handleSubmit} noValidate>
+                  {!serviceInfo?.makerRunning && !isWaitingMakerStart && !isWaitingMakerStop && (
+                    <>
+                      {settings.useAdvancedWalletMode && (
+                        <rb.Form.Group className="mb-3" controlId="offertype">
+                          <ToggleSwitch
+                            label={t('earn.toggle_rel_offer')}
+                            initialValue={values.offertype === OFFERTYPE_REL}
```

Comment:
Would something like `initialValue={values.offertype === initialValues.offertype}` here?

Reply:
Hmm.. I don't know if it'd be beneficial as the `onToggle` handler would also have to be adapted if the value of `initialValues.offertype` changed. The toggle is tailored towards being active if `offertype === OFFERTYPE_REL`. Does that make sense?